app/src/main/java/fr/free/nrw/commons/nearby/NearbyController.java

```diff
@@ -28,8 +28,10 @@
 public class NearbyController {
```

Comment:
Now that we have a presenter for Nearby, we should remove this class and shift the functionalities to the presenter

Reply:
Nearby controller controls our map logic (ie. how wide area is queried) while NearbyPresenter controls fragments and views. So I think we should keep them separated, but renaming NearbyController can be a better approach to eliminate confusion. What do you say @ashishkumar468 ?

Reply:
@neslihanturan any logic, be it controlling the map logic should go in the corresponding view presenter, we should not ideally have classes other than the presenter to have logics. If we talk in terms of unit testing, it would just be the presenter which would need to be tested.

Reply:
Hmm, but I can say that these two classes has really different responsibilities. Merging them not only makes the presenter class too long but also causes a semantic confusion. NearbyController decides which places will be displayed and what are their included information while presenter decides how to display them.

Reply:
The presenter is solely responsible for deciding how to display a view. If the view is complex then the presenter will become longer but the controller doesn't fit into the MVP semantics. 

If you are inclined towards having smaller presenters, you can try to break down the views into smaller logical views and have presenters for each of them.

Reply:
Can you please read what does NearbyController method do, it does not intended to control views, it does not decide how to display views either. It controls our search operation. It controls how we fetch data from Wikidata. Ie. storing our previously searched radius, our max number of results and accordingly giving us list of data in form of lets say Place type. It is not the same thing which is described in MVP documentation IMO. What do you say?